# I. Thông tin Dataset

## Ngữ cảnh:
Hiện nay với sự phát triển của ngành công nghiệp ô tô và sự phổ biến của ô tô trong đời sống, tiết kiệm nguyên liệu là một trong những tiêu chí quan trọng trong việc đánh giá một chiếc ô tô. Phân tích khả năng tiết kiệm nhiên liệu trong ngành công nghiệp ô tô mang lại nhiều lợi ích quan trọng, bao gồm:
- Giảm chi phí vận hành
- Bảo vệ môi trường
- Nâng cao tính cạnh tranh
- Phát triển công nghệ mới
- Đảm bảo an ninh năng lượng

Vì thế nên chúng em quyết định sẽ phân tích và đưa ra cái nhìn tổng quan chi tiết về khả năng tiết kiệm nhiên liệu trong ngành công nghiệp ô tô.
## Nguồn gốc
Dataset được lấy từ Kaggle (https://www.kaggle.com/datasets/tanishqdublish/vehcile-fuel-consumption/data)

## Nội dung
Bộ dữ liệu này cung cấp cái nhìn tổng quan chi tiết về khả năng tiết kiệm nhiên liệu trong ngành công nghiệp ô tô, bao gồm thông tin về nhiều loại xe từ các năm, nhãn hiệu và kiểu dáng khác nhau. Nó bao gồm các chi tiết về thông số kỹ thuật của động cơ, loại nhiên liệu, hệ thống truyền động cũng như các yếu tố như khả năng tiết kiệm nhiên liệu, lượng khí thải CO2 và tính năng công nghệ.

## License
- [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)

## Phương pháp thu thập
Dataset được tác giả lấy từ trang https://www.fueleconomy.gov/feg/ws/

Phương pháp:...

# II. Khám phá dữ liệu:

### Import thư viện cần thiết

Trước hết, import các thư viện cần thiết cho việc khám phá, xử lý và phân tích trở nên dễ dàng hơn.

In [13]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px

### Đọc dataset từ file csv
Đọc dữ liệu từ file `fuel.csv` được cung cấp

In [14]:
raw_data = pd.read_csv('../data/fuel.csv')

c:\Users\duypr\.conda\envs\min_ds-env\lib\site-packages\IPython\core\interactiveshell.py:3269: DtypeWarning: Columns (7,44) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


### In 5 dòng đầu tiên của dataset

In [15]:
raw_data.head()

vehicle_id  year        make               model  \
0       26587  1984  Alfa Romeo           GT V6 2.5   
1       27705  1984  Alfa Romeo           GT V6 2.5   
2       26561  1984  Alfa Romeo  Spider Veloce 2000   
3       27681  1984  Alfa Romeo  Spider Veloce 2000   
4       27550  1984  AM General   DJ Po Vehicle 2WD   

                         class          drive       transmission  \
0             Minicompact Cars            NaN     Manual 5-Speed   
1             Minicompact Cars            NaN     Manual 5-Speed   
2                  Two Seaters            NaN     Manual 5-Speed   
3                  Two Seaters            NaN     Manual 5-Speed   
4  Special Purpose Vehicle 2WD  2-Wheel Drive  Automatic 3-Speed   

  transmission_type  engine_index engine_descriptor  ...  \
0               NaN          9001             (FFS)  ...   
1               NaN          9005    (FFS) CA model  ...   
2               NaN          9002             (FFS)  ...   
3               NaN          9006    (FFS) CA model  ...   
4               NaN          1830             (FFS)  ...   

   hours_to_charge_ac_240v  composite_city_mpg composite_highway_mpg  \
0                      0.0                   0                     0   
1                      0.0                   0                     0   
2                      0.0                   0                     0   
3                      0.0                   0                     0   
4                      0.0                   0                     0   

   composite_combined_mpg range_ft1 city_range_ft1  highway_range_ft1  \
0                       0         0            0.0                0.0   
1                       0         0            0.0                0.0   
2                       0         0            0.0                0.0   
3                       0         0            0.0                0.0   
4                       0         0            0.0                0.0   

   range_ft2  city_range_ft2  highway_range_ft2  
0        NaN             0.0                0.0  
1        NaN             0.0                0.0  
2        NaN             0.0                0.0  
3        NaN             0.0                0.0  
4        NaN             0.0                0.0  

[5 rows x 81 columns]

### Số dòng và cột của dataset:

In [16]:
raw_data.shape

(38113, 81)

Như vậy bộ dữ liệu của chúng ta có 38113 dòng và 81 cột

### Ý nghĩa của mỗi dòng và cột
- Mỗi dòng mang ý nghĩa là những thông tin của mỗi một mẫu xe. Vì mỗi dòng mang thông tin của một mẫu xe khác nhau nên không có vấn đề nếu các dòng có ý nghĩa khác.
- Mỗi cột là mỗi một thuộc tính của mỗi chiếc xe.

### Các cột của dataset

In [17]:
np.array(raw_data.columns)

array(['vehicle_id', 'year', 'make', 'model', 'class', 'drive',
       'transmission', 'transmission_type', 'engine_index',
       'engine_descriptor', 'engine_cylinders', 'engine_displacement',
       'turbocharger', 'supercharger', 'fuel_type', 'fuel_type_1',
       'fuel_type_2', 'city_mpg_ft1', 'unrounded_city_mpg_ft1',
       'city_mpg_ft2', 'unrounded_city_mpg_ft2',
       'city_gasoline_consumption_cd', 'city_electricity_consumption',
       'city_utility_factor', 'highway_mpg_ft1',
       'unrounded_highway_mpg_ft1', 'highway_mpg_ft2',
       'unrounded_highway_mpg_ft2', 'highway_gasoline_consumption_cd',
       'highway_electricity_consumption', 'highway_utility_factor',
       'unadjusted_city_mpg_ft1', 'unadjusted_highway_mpg_ft1',
       'unadjusted_city_mpg_ft2', 'unadjusted_highway_mpg_ft2',
       'combined_mpg_ft1', 'unrounded_combined_mpg_ft1',
       'combined_mpg_ft2', 'unrounded_combined_mpg_ft2',
       'combined_electricity_consumption',
       'combined_gasoline_

### Lựa chọn các cột cần thiết

Vì phạm vi của project này chỉ là khám phá mối quan hệ trong dữ liệu, nên chúng ta không cần phải sử dụng tận 81 cột của dataset này. Ta chỉ cần lấy khoảng 15 thuộc tính đổ lại để phục vụ cho project này.

In [18]:
raw_data = raw_data[['vehicle_id', 'year', 'model', 'class', 'drive',
                    'transmission', 'engine_cylinders', 'engine_displacement',
                    'turbocharger', 'supercharger', 'fuel_type', 'fuel_type_1',
                    'city_mpg_ft1', 'highway_mpg_ft1', 'unadjusted_city_mpg_ft1']]

In [20]:
raw_data.head()

vehicle_id  year               model                        class  \
0       26587  1984           GT V6 2.5             Minicompact Cars   
1       27705  1984           GT V6 2.5             Minicompact Cars   
2       26561  1984  Spider Veloce 2000                  Two Seaters   
3       27681  1984  Spider Veloce 2000                  Two Seaters   
4       27550  1984   DJ Po Vehicle 2WD  Special Purpose Vehicle 2WD   

           drive       transmission  engine_cylinders  engine_displacement  \
0            NaN     Manual 5-Speed               6.0                  2.5   
1            NaN     Manual 5-Speed               6.0                  2.5   
2            NaN     Manual 5-Speed               4.0                  2.0   
3            NaN     Manual 5-Speed               4.0                  2.0   
4  2-Wheel Drive  Automatic 3-Speed               4.0                  2.5   

  turbocharger  supercharger fuel_type       fuel_type_1  city_mpg_ft1  \
0          NaN           NaN   Regular  Regular Gasoline            17   
1          NaN           NaN   Regular  Regular Gasoline            17   
2          NaN           NaN   Regular  Regular Gasoline            18   
3          NaN           NaN   Regular  Regular Gasoline            18   
4          NaN           NaN   Regular  Regular Gasoline            18   

   highway_mpg_ft1  unadjusted_city_mpg_ft1  
0               24                     21.0  
1               24                     21.0  
2               25                     23.0  
3               25                     23.0  
4               17                     22.0

### Ý nghĩa của các cột

### Ý nghĩa của các dòng

### Kiểm tra sự trùng lắp giữa các dòng

In [23]:
raw_data.duplicated().sum()

0

Như vậy dữ liệu của chúng ta không có sự trùng lắp giữa các dòng

### Kiểu dữ liệu của các cột

In [22]:
raw_data.dtypes

vehicle_id                   int64
year                         int64
model                       object
class                       object
drive                       object
transmission                object
engine_cylinders           float64
engine_displacement        float64
turbocharger                object
supercharger               float64
fuel_type                   object
fuel_type_1                 object
city_mpg_ft1                 int64
highway_mpg_ft1              int64
unadjusted_city_mpg_ft1    float64
dtype: object

###